src/dune/foreign.mli

```diff
@@ -70,6 +70,49 @@ val possible_sources :
   -> dune_version:Dune_lang.Syntax.Version.t
   -> string list
 
+module Archive : sig
+  module Name : sig
+    type t
+
+    module Map : sig
+      include Map.S with type key = t
+    end
+
+    val to_dyn : t -> Dyn.t
+
+    val to_string : t -> string
+
+    val path : dir:Path.Build.t -> t -> Path.Build.t
+
+    val decode : t Dune_lang.Decoder.t
+
+    val stubs : string -> t
+
+    val lib_file :
+      archive_name:t -> dir:Path.Build.t -> ext_lib:string -> Path.Build.t
+
+    val dll_file :
+      archive_name:t -> dir:Path.Build.t -> ext_dll:string -> Path.Build.t
+  end
+
+  module Dir : sig
+    type t
+  end
+
+  type t =
```

Comment:
You might as well this type abstract. Accessing `dir` isn't helpful, and you can always just have a function to access the name if it's necessary.

Reply:
Good idea, done.